src/functions.rs

```diff
@@ -1902,6 +1924,25 @@ impl<'a> FunctionInfo<'a> {
     pub fn get_declaration_range(&self) -> Option<Range<usize>> {
         self.decl.and_then(|d| d.name.get_range())
     }
+
+    // Convert a symbol defined by the arguments of this function into the symbol a given caller
+    // passed into it.  If the symbol is not the name of a function argument, return None
+    pub fn symbol_to_caller_symbol(
+        &self,
+        symbol: &str,
+        caller_args: &Vec<CilArg>,
+    ) -> Option<String> {
+        for (func_arg, caller_arg) in self.args.iter().zip(caller_args) {
+            if func_arg.name == symbol {
+                // We ignore perm lists, for no good reason.  This should probably return a CilArg
+                // and let the parent unpack it
```

Comment:
Worth opening an issue to track?

Reply:
Filed https://github.com/dburgener/cascade/issues/208